src/os/bluestore/KernelDevice.cc

```diff
@@ -256,6 +256,15 @@ void KernelDevice::_aio_thread()
 		 << " ioc " << ioc
 		 << " with " << left << " aios left" << dendl;
 	assert(r >= 0);
+
+	std::map<uint64_t, bufferlist*>::iterator it =  ioc->aio_read_queue.find(aio[i]->offset);
+	if (it != ioc->aio_read_queue.end()) {
+	  bufferlist *bl = it->second;
+	  bl->clear();
+	  bl->claim_append(aio[i]->bl);
+	  --ioc->num_reading;
+	}
+
```

Comment:
I think we need to add io type to FS::aio_t too. It looks like a little stupid to check aio_read_queue regardless the aio type.